src/test/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporterTest.java

```diff
@@ -289,6 +291,56 @@ public void shouldReportExpectedHistogramStdDevAsIs() throws Exception {
     }
 
     @Test
+    public void shouldReportExpectedHistogramMaxMinAsIs() throws Exception {
+        metricRegistry.histogram("TheHistogram").update(1);
+        metricRegistry.histogram("TheHistogram").update(2);
+        metricRegistry.histogram("TheHistogram").update(3);
+        metricRegistry.histogram("TheHistogram").update(30);
+        reporterBuilder.withStatisticSet().build().report();
+
+        final MetricDatum metricData = metricDatumByDimensionFromCapturedRequest("snapshot-summary");
+
+        assertThat(metricData.getStatisticValues().getMaximum().intValue()).isEqualTo(30);
+        assertThat(metricData.getStatisticValues().getMinimum().intValue()).isEqualTo(1);
+        assertThat(metricData.getUnit()).isEqualTo("None");
+    }
+
+    @Test
+    public void shouldReportSubsequentMaxMinValues() throws Exception {
+        Histogram slidingWindowHistogram = new Histogram(new SlidingWindowReservoir(4));
+        metricRegistry.register("SlidingWindowHistogram", slidingWindowHistogram);
+
+        slidingWindowHistogram.update(1);
+        slidingWindowHistogram.update(2);
+        slidingWindowHistogram.update(30);
+        CloudWatchReporter reporter = reporterBuilder.withStatisticSet().build();
+        reporter.report();
+
+        final MetricDatum metricData = metricDatumByDimensionFromCapturedRequest("snapshot-summary");
+
+        assertThat(metricData.getStatisticValues().getMaximum().intValue()).isEqualTo(30);
+        assertThat(metricData.getStatisticValues().getMinimum().intValue()).isEqualTo(1);
+        assertThat(metricData.getStatisticValues().getSampleCount().intValue()).isEqualTo(3);
+        assertThat(metricData.getStatisticValues().getSum().intValue()).isEqualTo(33);
+        assertThat(metricData.getUnit()).isEqualTo("None");
+
+        slidingWindowHistogram.update(4);
+        slidingWindowHistogram.update(100);
+        slidingWindowHistogram.update(5);
+        slidingWindowHistogram.update(6);
+        reporterBuilder.withStatisticSet().build().report();
```

Comment:
I am curious: did you intentionally re-build the `CloudWatchReporter` before reporting the 2nd time? Why not just `reporter.report();`?